src/Workspaces/CSharp/Portable/Editing/CSharpImportAdder.cs

```diff
@@ -66,138 +66,102 @@ protected override SyntaxNode MakeSafeToAddNamespaces(SyntaxNode root, IEnumerab
+                base.VisitIdentifierName(node);
+                CheckName(node);
             }
 
-            public override SyntaxNode VisitInvocationExpression(InvocationExpressionSyntax node)
+            public override void VisitGenericName(GenericNameSyntax node)
             {
-                // No need to visit trivia, as we only care about xml doc comments
-                if (node.Expression is MemberAccessExpressionSyntax memberAccess)
-                {
-                    if (_extensionMethods.Contains(memberAccess.Name.Identifier.Text))
-                    {
-                        // No need to visit this as simplifier will expand everything
-                        return Simplifier.Expand<SyntaxNode>(node, _model, _workspace, cancellationToken: _cancellationToken);
-                    }
-                }
-
-                return base.VisitInvocationExpression(node) ?? throw ExceptionUtilities.Unreachable;
+                base.VisitGenericName(node);
+                CheckName(node);
             }
 
-            public override SyntaxNode VisitMemberAccessExpression(MemberAccessExpressionSyntax node)
+            public override void VisitMemberAccessExpression(MemberAccessExpressionSyntax node)
             {
-                node = (MemberAccessExpressionSyntax)(base.VisitMemberAccessExpression(node) ?? throw ExceptionUtilities.Unreachable);
+                base.VisitMemberAccessExpression(node);
 
-                if (_extensionMethods.Contains(node.Name.Identifier.Text))
-                {
-                    // If an extension method is used as a delegate rather than invoked directly,
-                    // there is no semantically valid transformation that will fully qualify the extension method. 
-                    // For example `Func<int> f = x.M;` is not the same as Func<int> f = () => Extensions.M(x);`
-                    // since one captures x by value, and the other by reference.
-                    //
-                    // We will not visit this node if the parent node was an InvocationExpression, 
-                    // since we would have expanded the parent node entirely, rather than visiting it.
-                    // Therefore it's possible that this is an extension method being used as a delegate so we warn.
-                    node = node.WithAdditionalAnnotations(WarningAnnotation.Create(string.Format(
-                        WorkspacesResources.Warning_adding_imports_will_bring_an_extension_method_into_scope_with_the_same_name_as_member_access,
-                        node.Name.Identifier.Text)));
-                }
+                // Check to see if we have a reference to an extension method.  If so, then pulling in an import could
+                // bring in an extension that conflicts with that.
 
-                return node;
-            }
+                var symbol = _model.GetSymbolInfo(node.Name, _cancellationToken).GetAnySymbol();
```

Comment:
Even if it's not an extension method, this could still lead to ambiguities in lambdas, where you could now make multiple types for the parameter compile where only one could before.

Reply:
Could you give an example of what you mean here?

Reply:
```csharp
using System;
class C
{
    public static void Main()
    {
        M(x => x.M1());
    }

    public static void M(Action<C> a){}
    public static void M(Action<int> a){}

    public void M1(){}

}

namespace N
{
    public static class Extensions
    {
        public static void M1(this int a){}
    }
}
```

Importing `N` will make this ambiguous.

Reply:
Yup.  In this case, i think an appropriate course of action is to do the check for extension-method collisions normally, but check for *all* method collisions when in a lambda.  The effective rule here would be that since lambdas may be re-bound in different resolution contexts, we can't know if that instance method might be reinterpreted as an extension with a different inference context.

Reply:
This case is handled now.